lib/active_record/connection_adapters/odbc_adapter.rb

```diff
@@ -50,15 +51,20 @@ class ODBCAdapter < AbstractAdapter
 
       def initialize(config_or_deprecated_connection, deprecated_logger = nil, deprecated_connection_options = nil, deprecated_config = nil, database_metadata = nil)
         super(config_or_deprecated_connection, deprecated_logger, deprecated_connection_options, deprecated_config)
-        @raw_connection = config_or_deprecated_connection
-
-        if database_metadata
-          @database_metadata = database_metadata
+        @config = deprecated_config
+        if config_or_deprecated_connection.try(:get_info, ODBC.const_get("SQL_DBMS_NAME"))
+          @raw_connection = config_or_deprecated_connection
+          @config = deprecated_config
+          configure_time_options(@raw_connection)
         else
-          @database_metadata = ::ODBCAdapter::DatabaseMetadata.new(@raw_connection)
+          config = config_or_deprecated_connection
+          setup = ::ODBCAdapter::ConnectionSetup.new(config.symbolize_keys)
+          setup.build
+          @config = setup.config
```

Comment:
Why do we not need the time options on this side of the if test?

Reply:
`connect` (on the next line) calls `configure_time_options`